egui/src/context.rs

```diff
@@ -464,33 +464,41 @@ impl Context {
 
     // ---------------------------------------------------------------------
 
-    /// Constraint the position of a window/area
+    /// Constrain the position of a window/area
     /// so it fits within the screen.
     pub(crate) fn constrain_window_rect(&self, window: Rect) -> Rect {
-        let mut screen = self.available_rect();
+        self.constrain_rect_impl(window, self.available_rect())
+    }
+
+    /// Constrain the position of a window/area
+    /// so it fits within the provided boundary.
+    pub(crate) fn constrain_window_rect_to_area(&self, window: Rect, area: Rect) -> Rect {
+        self.constrain_rect_impl(window, area)
+    }
 
-        if window.width() > screen.width() {
+    fn constrain_rect_impl(&self, window: Rect, mut area: Rect) -> Rect {
```

Comment:
no need for both `constrain_rect_impl` and `constrain_window_rect_to_area`. Just keep `constrain_window_rect_to_area` and call it from `constrain_window_rect`